UefiCpuPkg/Library/MpInitLib/MpLib.c

```diff
@@ -1643,14 +1643,30 @@ TimedWaitForApFinish (
                               TimeLimit,
                               &CpuMpData->CurrentTime
                               );
-  while (CpuMpData->FinishedCount < FinishedApLimit &&
-         !CheckTimeout (
-            &CpuMpData->CurrentTime,
-            &CpuMpData->TotalTime,
-            CpuMpData->ExpectedTime
-            ))
   {
-    CpuPause ();
+    UINTN  LoopCount;

```

Comment:
Please move these declarations to the start of the function, and drop the redundant enclosing {}